MilitaryManager.Core/DTO/Units/UnitDTO.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+
+namespace MilitaryManager.Core.DTO.Units
+{
+    public class UnitDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Address { get; set; }
+        public int? ParentId { get; set; }
+
+        public virtual UnitDTO Parent { get; set; }
+
+        public virtual ICollection<UnitDTO> SubUnits { get; set; }
```

Comment:
What difference between virtual and not virtual navigation properties 
Are we sure that we need to use virtual properties